src/main/kotlin/net/maizegenetics/phgv2/pathing/AlignmentUtils.kt

```diff
@@ -0,0 +1,292 @@
+
+
+    val rangeToBitSetMap = mutableMapOf<ReferenceRange, BitSet>()
+    val kmerHashMap = Long2LongOpenHashMap()
+    var lineCount = 0
+    var totalLineCount = 0
+    var refrange = ReferenceRange("NULL", 0, 0)
+
+    val refRangeToIdMap = getReferenceRangeToIndexMap(graph)
+    getBufferedReader(filename).useLines {
+        it.forEach { inputStr ->
+            totalLineCount++
+
+            lineCount = when(inputStr.first()) {
+                '>' -> 1
+                else -> lineCount + 1
+            }
+
+            when (lineCount) {
+                1 -> {
+                    //line 1 is the range
+                    refrange = ReferenceRange.parse(inputStr.substring(1))
+                }
+                2-> {
+                    try {
+                        val parsedLine = inputStr.split(",")
+                        val myBitset = BitSet.valueOf(parsedLine.map { it.toLong() }.toLongArray())
+                        rangeToBitSetMap[refrange] = myBitset
+                    } catch(e: Exception) {
+                        println("error at line $totalLineCount for input = $inputStr")
+                        throw java.lang.IllegalArgumentException(e)
+                    }
+                }
+                3 -> {
+                    val refrangeLong = (refRangeToIdMap[refrange]!!.toLong()) shl 32
+                    val parsedLine = inputStr.split(",")
+                    //values are pairs of hash, offset (long,int)
+                    //add the pairs of entries to kmerHashMap
+                    for (datapair in parsedLine) {
+                        val hashOffset = datapair.split("@")
+
+                        if (hashOffset.size < 2) {
+                            println("improperly formatted datapair at line $totalLineCount")
+                        }
+
+                        val hash = hashOffset[0].toLong()
+                        val offset = refrangeLong or hashOffset[1].toLong()
+                        kmerHashMap.put(hash, offset)
+                    }
+                }
+                else -> {
+                    throw IllegalArgumentException("Line count = $lineCount in kmerMap file format at line $totalLineCount in file $filename")
+                }
+            }
+        }
+    }
+
+    return KmerMapData(rangeToBitSetMap, kmerHashMap)
+}
+
+/**
+ * Creates a map of ReferenceRange -> (map of hapid -> index) for a [HaplotypeGraph]
+ */
+fun getRefRangeToHapidMap(graph: HaplotypeGraph) : Map<ReferenceRange,Map<String,Int>>{
+    //This creates a map of ReferenceRangeId -> (map of hapid -> index)
+    return graph.ranges().associateWith { range ->
+        graph.hapIdToSamples(range).keys.toSortedSet()
+            .mapIndexed { index, hapid -> hapid to index }.toMap()
+    }
+}
+
+/**
+ * Creates a map of ReferenceRange -> index for a [HaplotypeGraph]. Because the ranges are sorted by the
+ * HaplotypeGraph method range(), a graph always returns the same map.
```

Comment:
In "sorted by the HaplotypeGraph method range()".  What is method here?

Reply:
I am a bit confused.  It is calling the graph.ranges() method.  There is a typo because it is ranges() not range() so I have fixed it.